pretyping/evarconv.ml

```diff
@@ -933,28 +934,6 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) flags env evd pbty
           in
             ise_try evd [f1; f2]
 
-        (* Catch the p.c ~= p c' cases *)
-        | Proj (p,_,c), Const (p',u) when QConstant.equal env (Projection.constant p) p' ->
-          let res =
-            try Some (destApp evd (Retyping.expand_projection env evd p c []))
-            with Retyping.RetypeError _ -> None
-          in
-            (match res with
-            | Some (f1,args1) ->
-              evar_eqappr_x flags env evd pbty (f1,Stack.append_app args1 sk1)
-                appr2
-            | None -> UnifFailure (evd,NotSameHead))
-
-        | Const (p,u), Proj (p',_,c') when QConstant.equal env p (Projection.constant p') ->
-          let res =
-            try Some (destApp evd (Retyping.expand_projection env evd p' c' []))
-            with Retyping.RetypeError _ -> None
-          in
-            (match res with
-            | Some (f2,args2) ->
-              evar_eqappr_x flags env evd pbty appr1 (f2,Stack.append_app args2 sk2)
-            | None -> UnifFailure (evd,NotSameHead))
-
```

Comment:
Since compat constants are now transparent by default this case and the `Rigid, Rigid` one below would only trigger for when using `Opaque!` (the new way to force transparency of compat constants). Do we still want these cases so that unification works fine in that scenario?

Reply:
using ! means t should fail MO

Reply:
It turns out these lines were useful in some cases (see the commit by @mattam82 that introduced them: 6bca54599ab2b8ab928bfc92b8ddfb0aeba4345f). Without this, `test-suite/bugs/bug_3484.v` fails. The unification debug log is as follows prior to re-adding the lines:
```
[unification]
  projT1|ZApp(T, fun g : T => @paths T g g, ?y)
  @projT1 _ _ (@existT T (fun g : T => @paths T g g) x (@idpath T x))|
 
[unification]
  projT1|ZApp(T, fun g : T => @paths T g g, ?y)
  x|
 
[unification]
  @projT1 _ _ ?y|
  x|

[unification]
  ?y|ZProj(unfolded((bug_3484.projT1)))
  x|
```
And on `master` it is:
```
[unification]
  projT1|ZApp(T, fun g : T => @paths T g g, ?y)
  @projT1 _ _ (@existT T (fun g : T => @paths T g g) x (@idpath T x))|

[unification]
  projT1|ZApp(T, fun g : T => @paths T g g, ?y)
  projT1|ZApp(T, fun g : T => @paths T g g, @existT T (fun g : T => @paths T g g) x (@idpath T x))
```

Reply:
If we wanted to experiment further we could even try to replace the 1st case by `| _, Const (p, _) when Option.has_some (PrimitiveProjections.find_opt p) -> .. unfolds the constant ...` (and similarly for the 2nd case) but it seems to be that it is also good to keep it as it is (until a clear policy is taken for Constant encapsulating primitive projections).